File: seed/rust-model/exhaustive/src/error_category.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub enum ErrorCategory {
    #[serde(rename = "API_ERROR")]
    ApiError,
    #[serde(rename = "AUTHENTICATION_ERROR")]
    AuthenticationError,
    #[serde(rename = "INVALID_REQUEST_ERROR")]
    InvalidRequestError,
}

File: seed/ruby-sdk-v2/exhaustive/wire-tests/lib/fern_exhaustive/endpoints/put/types/error_category.rb
# frozen_string_literal: true

module FernExhaustive
  module Endpoints
    module Put
      module Types
        module ErrorCategory
          extend FernExhaustive::Internal::Types::Enum

          API_ERROR = "API_ERROR"
          AUTHENTICATION_ERROR = "AUTHENTICATION_ERROR"
          INVALID_REQUEST_ERROR = "INVALID_REQUEST_ERROR"
        end
      end
    end
  end
end
